src/meta/src/stream/stream_manager.rs

```diff
@@ -531,6 +536,15 @@ where
                 .await?;
         }
 
+        // Register to compaction group beforehand.
+        // If any following operation fails, the registration will be eventually reverted.
+        //
+        // Note that this step must be before building actors, because we will explicitly sync this
+        // info in the beginning of actors.
+        self.compaction_group_manager
+            .register_table_fragments(&table_fragments, &table_properties)
+            .await?;
+
         // In the second stage, each [`WorkerNode`] builds local actors and connect them with
         // channels.
         for (node_id, actors) in node_actors {
```

Comment:
Do we need to `unregister_table_fragments` if BuildActorsRequest fails?

Reply:
`purge_stale_members` ensures stale data is eventually cleaned up on each restart.
We can unregister_table_fragments/unregister_source explicitly to clean up more timely. I don't have it here, in order to avoid introducing more error handling code here.

Reply:
fixed